Dockerfile

```diff
@@ -52,12 +59,21 @@ ENTRYPOINT [ "solvers" ]
 
 # Extract Binary
 FROM intermediate
+RUN apt-get update && \
```

Comment:
I have multiple concerns here:
1. when we don't need heaptrack we rebuild it over and over again in the deploy action for no use. To get around this it could be made opt-in with a flag. Alternatively we could cache building heaptrack somehow. That way we only pay with increased image size which would probably be relatively small.
2. heaptracking should be available to all binaries. Sure currently we only have an issue in the orderbook pod but we could very well have an issue in the autopilot pod at some other time
3. the infra should be able to decide whether to run the instrumented version or not at runtime
4. building `heaptrack` right in the final image means we have a bunch of dependencies we only need for building but not for running heaptrack. It should be built in a separate container and we should only copy over the binaries and libs we actually need.

Reply:
This PR will not be merged. It can be tackled once we decide to deal with the heaptrack on a long-term basis.